flink-libraries/flink-table/src/main/scala/org/apache/flink/table/functions/utils/ScalarSqlFunction.scala

```diff
@@ -136,8 +136,18 @@ object ScalarSqlFunction {
       }
 
       override def getOperandCountRange: SqlOperandCountRange = {
-        val signatureLengths = signatures.map(_.length)
-        SqlOperandCountRanges.between(signatureLengths.min, signatureLengths.max)
+        var min = 255
+        var max = -1
+        signatures.foreach(sig => {
+          var len = sig.length
+          if (len > 0 && sig(sig.length -1).isArray) {
+            max = 254  // according to JVM spec 4.3.3
```

Comment:
Hi @wuchong . Yes, according to JVM specification. It may have 255 parameters, if the method is static. If the method is not static, the pointer "this" will be one of the parameters, so it is 254.

FYI, http://docs.oracle.com/javase/specs/jvms/se7/html/jvms-4.html#jvms-4.3.3